File: WayCare/src/main/java/com/example/waycare/models/Localizacao.java
package com.example.waycare.models;
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "localizacao")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Localizacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "loc_id")
    private Long id;

    @Column(name = "loc_latitude")
    private Double latitude;

    @Column(name = "loc_longitude")
    private Double longitude;

    @Column(name="loc_endereco")
    private String endereco;

    @OneToMany(mappedBy = "localizacao")
    @JsonBackReference
    private List<Reporte> reporte;
}

File: frontend/app/src/main/java/pt/iade/ei/waycareapp/data/model/Localizacao.kt
package pt.iade.ei.waycareapp.data.model

import com.google.gson.annotations.SerializedName

data class Localizacao(
    @SerializedName("id")
    val loc_id: Long? = null,

    @SerializedName("latitude")
    val loc_latitude: Double,

    @SerializedName("longitude")
    val loc_longitude: Double,

    @SerializedName("endereco")
    val loc_endereco: String
)
